integrations/aws/client.go

```diff
@@ -86,10 +93,39 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
 	ConnectionUrl: "/aws/connect",
 }))
 
-func New(vars sdkservices.Vars) sdkservices.Integration {
+func New(cvars sdkservices.Vars) sdkservices.Integration {
+	i := &integration{vars: cvars}
 	return sdkintegrations.NewIntegration(
 		desc,
-		sdkmodule.New(initOpts(vars)...),
-		sdkintegrations.WithConnectionConfigFromVars(vars),
+		sdkmodule.New(initOpts(cvars)...),
+		connStatus(i),
+		sdkintegrations.WithConnectionConfigFromVars(cvars),
 	)
 }
+
+// connStatus is an optional connection status check provided by
+// the integration to AutoKitteh. The possible results are "init
+// required" (the connection is not usable yet) and "initialized".
+func connStatus(i *integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionStatus(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
+		}
+
+		vs, err := i.vars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		at := vs.Get(authType)
+		if !at.IsValid() || at.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
+		}
+
+		if at.Value() == integrations.Init {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "initialized"), nil
+		} else {
```

Comment:
According Go idiomatic style, no need for `else` when both branches simply `return` - use this:

```go
if ... {
    return ...
}
return ...
```

...And in other files too

Reply:
Done